mysql_tests/test_crud.py

```diff
@@ -0,0 +1,333 @@
+    assert not f2
+
+
+async def test_string_primary_key(engine):
+    relations = ["Colleagues", "Friends", "Lovers"]
+    for r in relations:
+        await Relation.create(bind=engine, timeout=10, name=r)
+    r1 = await Relation.get(relations[0], bind=engine, timeout=10)
+    assert r1.name == relations[0]
+
+
+async def test_lookup_287(bind):
+    from gino.exceptions import NoSuchRowError
+
+    class Game(db.Model):
+        __tablename__ = "games"
+        game_id = db.Column(db.String(32), unique=True)
+        channel_id = db.Column(db.String(1), default="A")
+
+    await Game.gino.create()
+    try:
+        game_1 = await Game.create(game_id="1", channel_id="X")
+        game_2 = await Game.create(game_id="2", channel_id="Y")
+
+        # ordinary update should be fine
+        uq = game_1.update(game_id="3")
+
+        with pytest.raises(TypeError, match="Model Game has no table, primary key"):
+            # but applying the updates to DB should fail
+            await uq.apply()
+
+        with pytest.raises(
+            LookupError, match="Instance-level CRUD operations not allowed"
+        ):
+            await game_2.delete()
+        with pytest.raises(
+            LookupError, match="Instance-level CRUD operations not allowed"
+        ):
+            await game_2.query.gino.all()
+        with pytest.raises(
+            LookupError, match="Instance-level CRUD operations not allowed"
+        ):
+            await game_2.select("game_id")
+
+        # previous ordinary update still in effect
+        assert game_1.game_id == "3"
+
+        assert await Game.select("game_id").gino.all() == [("1",), ("2",)]
+
+        Game.lookup = lambda self: Game.game_id == self.game_id
+        with pytest.raises(NoSuchRowError):
+            await game_1.update(channel_id="Z").apply()
+        await game_2.update(channel_id="Z").apply()
+        assert await Game.select("channel_id").gino.all() == [("X",), ("Z",)]
+    finally:
+        await Game.gino.drop()
+
+
+async def test_lookup_custom_name(bind):
+    class ModelWithCustomColumnNames(db.Model):
+        __tablename__ = "gino_test_custom_column_names"
+
+        id = db.Column("other", db.Integer(), primary_key=True)
+        field = db.Column(db.Text())
+
+    await ModelWithCustomColumnNames.gino.create()
+
+    try:
+        # create
+        m1 = await ModelWithCustomColumnNames.create(id=1, field="A")
+        m2 = await ModelWithCustomColumnNames.create(id=2, field="B")
+
+        # update
+        uq = m1.update(field="C")
+        await uq.apply()
+
+        # lookup
+        assert set(
+            tuple(x) for x in await ModelWithCustomColumnNames.select("id").gino.all()
+        ) == {(1,), (2,)}
+        assert (await ModelWithCustomColumnNames.get(2)).field == "B"
+        assert (await ModelWithCustomColumnNames.get(1)).field == "C"
+        assert await ModelWithCustomColumnNames.get(3) is None
+
+        # delete
+        assert (
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/python-gino/gino/pullRequest?prid=5587950)